File: cpp/dfs.cpp
#include<stack>
#include<iostream>

using namespace std;

int[] dfs(int* arr, int start, bool[] visited)
{
    visited[start] = true;
    stack<int> st;
    st.push(start);
    while (!st.empty)
    {
        int var = st.pop();
        for(int i = 0; i < arr[var].length; i++)
        {
            if (visited[])
            {
                /* code */
            }
            

        }
        
    }
    
    
}

int main(void)
{

}

File: python/Old/dfs.py


def dfs(grpah, start, visited):
    st = [start]

    while(st):
        v = st.pop()
        visited[v] = True
        print(v, end=" ")
        for i in grpah[v]:
            if not visited[i] and i not in st:
                st.append(i)


graph = [
    [],
    [2, 3, 4, 6],
    [1, 4],
    [1, 5],
    [1, 2, 5],
    [3, 4],
    [1]
]

visited = [False] * 7
visited[0] = True

dfs(graph, 1, visited)